include/swift/Driver/Compilation.h

```diff
@@ -258,6 +262,22 @@ class Compilation {
     }
   }
 
+  /// Returns whether the callee is permitted to pass -emit-tbd to a frontend
+  /// job.
+  ///
+  /// This only returns true once, because only one job should pass that
+  /// argument.
+  bool requestPermissionForFrontendToEmitTBD() {
+    if (PassedEmitTBDToFrontendJob)
+      // Someone else has already done it!
+      return false;
```

Comment:
Although actually I don't think this even makes sense. A -primary-file job won't have typechecked any of the declarations in the other files. We should just limit it to -whole-module-optimization / -force-single-frontend-invocation mode for now.